### DISCUSSION TITLE
除了copilot和chatGPT以外，也能用AI寫程式的免費套件，Tabnine

### DISCUSSION THREAD

**Discussion by williamyen0623** [1101-GS3073-C/Python-Community-GS3073-C#560] [科技新知]
https://blog.darkthread.net/blog/tabnine/
這個網站主要是在介紹Tabnine

使用方法相當簡單，只要到vscode的extension下載安裝就好，然後按照步驟去官網註冊
比較逗趣的是，我的vscode同時開著copilot跟Tabnine，常常不知道到底是哪個套件幫我寫的程式哈哈
一直按tab一直爽

**Comment by HsuChen88** [comment]
我想也是時候嘗試新工具了，AI寫的code可能比我還好，飯碗不保。

**Comment by skyling0299** [comment]
Tabnine通過自動完成他們的代碼來提高開發人員的工作效率。 智能代碼完成可節省大量時間。 在您鍵入時，Tabnine 會根據上下文和語法預測並建議您的下一行代碼。(測試機器人用)

**Reply by skyling0299** [reply]
這是關於我的另一個討論「PyPal機器人如何判斷原創性」的測試內容~
如果你有興趣，歡迎加入測試行列~~
